piecrust-uplink/src/types.rs

```diff
@@ -118,20 +134,25 @@ impl RawQuery {
         Self { arg_len, data }
     }
 
-    pub fn name_bytes(&self) -> &[u8] {
-        &self.data[self.arg_len as usize..]
-    }
-
+    /// Return a reference to the name of [`RawQuery`]
     pub fn name(&self) -> &str {
         core::str::from_utf8(self.name_bytes())
             .expect("always created from a valid &str")
     }
 
+    /// Return a reference to the raw name of [`RawQuery`]
+    pub fn name_bytes(&self) -> &[u8] {
+        &self.data[self.arg_len as usize..]
+    }
+
+    /// Return a reference to the raw argument of [`RawQuery`]
     pub fn arg_bytes(&self) -> &[u8] {
         &self.data[..self.arg_len as usize]
     }
 }
 
+/// A RawTransaction is a host transaction that doesn't care about types and
+/// only operates on raw data.
```

Comment:
Again, not a "host transaction". This nomenclature should be reserved for actual `HostQuery`es.